backend/src/main/java/com/soen341/instagram/model/Likable.java

```diff
@@ -0,0 +1,29 @@
+package com.soen341.instagram.model;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.persistence.ManyToMany;
+import javax.persistence.MappedSuperclass;
+
+@MappedSuperclass
+public abstract class Likable {
+	@ManyToMany
+	private Set<Account> likedBy;
+	
+	public Likable() {
+		likedBy = new HashSet<Account>();
+	}
+	
+	public Set<Account> getLikedBy() {
+		if (likedBy.equals(null))
+			return new HashSet<Account>();
```

Comment:
should be:

```
this.likedBy = new HashSet<Account>();
return this.likedBy;
```

Otherwise, let's say that `likedBy` is null, then doing this:
```
Comment c = new Comment();
c.getLikedBy().add(user);
repo.save(c);
```
Will not save the user's like